### PR TITLE
Add `!=` operator for version resolve

### LINKED ISSUE #519: Add support for != operator in Crystal version
Sometimes upstream Crystal bugs can introduce backward incompatible changes. IE crystal-lang/crystal#10972. It may be useful to have support for a `!=` operator to negate specific versions of crystal.

### PR DESCRIPTION
**Pull request by syeopite**
Closes #519

### CODE REVIEW

**Comment by beta-ziliani** [comment]
The test is failing because of the formatter. Let me know @syeopite if you want me to take care of it, or if you can fix it.

**Comment by syeopite** [comment]
Ah thanks! I couldn't figure out why CI was failing.

**Review by beta-ziliani — APPROVED** [review]
Once CI says so, I say this is ready to go. Thanks!

File: spec/unit/versions_spec.cr [hunk 1]
```diff
@@ -147,6 +147,13 @@ module Shards
       resolve(versions, "0.0.1.alpha").should be_empty
     end
 
+    it "resolves neq" do
+      versions = %w(0.0.1 0.1.0 0.1.1 0.1.2 0.2.0 0.10.0)
+
+      resolve(versions, "!= 0.1.0").should eq(["0.0.1", "0.1.1", "0.1.2", "0.2.0", "0.10.0"]),
+        resolve(versions, "!= 0.1.0, != 0.2.0, != 0.10.0").should eq(["0.0.1", "0.1.1", "0.1.2"])
```
Comment (Sija) [inline spec/unit/versions_spec.cr:154]:
```suggestion
      resolve(versions, "!= 0.1.0").should eq(["0.0.1", "0.1.1", "0.1.2", "0.2.0", "0.10.0"])
      resolve(versions, "!= 0.1.0, != 0.2.0, != 0.10.0").should eq(["0.0.1", "0.1.1", "0.1.2"])
```

**Review by Sija — CHANGES_REQUESTED** [review]

File: spec/unit/versions_spec.cr [hunk 2]
```diff
@@ -199,6 +206,11 @@ module Shards
       matches?("1.0.0", "1.0").should be_true
       matches?("1.0.0", "1.0.1").should be_false
 
+      matches?("1.0.0", "> 0.1.0, < 1.0.1, != 1.0.0").should be_false
+      matches?("1.0.0", "> 0.1, < 1.0, != 1.0").should be_false
```
[the remark this answers is not in the document]
Comment (syeopite) [reply spec/unit/versions_spec.cr]: Yeah, I've noticed that a few days ago and fixed it in commit 30df126

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**